readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/kafka/OffsetResetter.java

```diff
@@ -52,41 +50,54 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
 
     @Override
     public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
-         for (TopicPartition partition : partitions) {
-            long startingOffset = UNKNOWN_OFFSET;
-            List<Long> offsets = new ArrayList<>(writers.size());
+        Map<Integer, Collection<Long>> offsetsPerPartition = new HashMap<>();
```

Comment:
I think it could be good to explain here that the way the code is shapped is to collect offsets in a fast manner to prevent kafka from considering the consumer as dead.